### PR TITLE
🐛 Source FacebookPages: Fixed `access_token` key error

### PR DESCRIPTION
**Pull request by lgomezm**
## What
* Fixes error when checking connection: `KeyError: 'access_token'` (https://github.com/airbytehq/airbyte/issues/9349).
* Fixes another issue with deprecated page field: `nativeoffers` (https://github.com/airbytehq/airbyte/issues/13756).

## How
* Uses the Facebook Graph version to latest: 14.0.
* Removes expected field that is deprecated: `nativeoffers`.
* Adds better error handling at source's `check_connection`.

## Recommended reading order
1. `source.py`
2. `streams.py`
3. `metrics.py`
4. `unit_test.py`
5. `setup.py`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

Screen Shot 2022-07-12 at 10 41 17 PM

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-facebook-pages/unit_tests/unit_test.py [hunk 1]
```diff
@@ -16,6 +21,49 @@ def json(self):
...
+
+        assert ok
+        assert not error_msg
+
+
+def test_check_connection_empty_config():
+    ok, error_msg = SourceFacebookPages().check_connection(logger, config={})
+
+    assert not ok
+    assert error_msg
```
Comment (itaseskii) [inline airbyte-integrations/connectors/source-facebook-pages/unit_tests/unit_test.py:49]: Consider asserting concrete error messages to distinguish between different error scenarios.

Same applies to 

```
test_check_connection_invalid_config(config)
test_check_connection_exception(config)
```

**Comment by lgomezm** [comment]
@marcosmarxm Integration tests are failing because the credentials used don't have enough permissions. I can see this HTTP 400 error in the logs: `Requires pages_read_user_content permission to manage the object`. Could they be updated?

File: airbyte-integrations/connectors/source-facebook-pages/unit_tests/unit_test.py (hunk 1 above)
Comment (lgomezm) [reply airbyte-integrations/connectors/source-facebook-pages/unit_tests/unit_test.py]: Good deal. Updated in [96d63b8](https://github.com/airbytehq/airbyte/pull/14660/commits/96d63b8fbf8ae93b540b7d6a158f32be7795c3ff).

**Comment by linuxs** [comment]
I'm blocked from adding FacebookPages sources within Airbyte a the moment due to `Removes expected field that is deprecated: nativeoffers.`. It would be great if we could ship a portion of this PR to unblock this.

**Comment by natalyjazzviolin** [comment]
@linuxs @lgomezm we are working on adding the correct permissions so the PR can be tested and merged. Thanks for your patience!

**Comment by marcosmarxm** [comment]
@natalyjazzviolin any updates here?

**Comment by natalyjazzviolin** [comment]
@marcosmarxm we still don't have the correct permissions to run the tests, unfortunately.

**Comment by lgomezm** [comment]
@marcosmarxm @natalyjazzviolin Is there anything else that could be done to get this PR merged?

**Comment by YowanR** [comment]
@natalyjazzviolin Could you share details on why we don't have the right permissions? Is this a restrictions from Fb? Happy to help unblock here if I can.

**Comment by schickler-o-martin** [comment]
Any update on this PR? We would really like to use this connector as well ;-)

**Comment by natalyjazzviolin** [comment]
> @marcosmarxm Integration tests are failing because the credentials used don't have enough permissions. I can see this HTTP 400 error in the logs: `Requires pages_read_user_content permission to manage the object`. Could they be updated?

@YowanR we need the permissions Marcos mentioned above: `pages_read_user_content` permissions. Are you able to grant these?

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

**Comment by ycherniaiev** [comment]
@YowanR To receive the reqiired permission we need provide the FB team with:
1. Account for testing with detailed instructions how this scope is used
2. Demo Video of how this scope is used
Second - is possible in local (since FB pages not on cloud)
First - is not possible

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

**Comment by marcosmarxm** [comment]
Closed because facebook page now uses low-code CDK https://github.com/airbytehq/airbyte/pull/19788 Thanks forthe contribution @lgomezm

**Pull request closed without merging**